bin/git-rename-file

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# Usage function
+usage() {
+    cat <<EOF
+Usage: git rename-file [OPTIONS] <source> <destination>
+
+Description:
+  Rename a file or directory and ensure Git recognizes the change, regardless of filesystem case-sensitivity.
+  It combines the functionality of the "mv" command and "git mv". This is particularly useful for renaming files or directories
+  to change only their case, which might not be detected by Git on case-insensitive filesystems.
+
+Options:
+  -h, --help  Show this help message and exit.
+
+Examples:
+  git rename-file old_filename new_filename
+  git rename-file old_directory new_directory
+EOF
+}
+
+# Check for help option
+if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+    usage
+    exit 0
+fi
+
+# Check for correct number of arguments
+if [ "$#" -ne 2 ]; then
+    echo "Error: Incorrect number of arguments."
+    echo ""
+    usage
+    exit 1
+fi
+
+# Assign variables
+SOURCE="$1"
+DESTINATION="$2"
+TEMP_NAME="${SOURCE}.temp"
+
+# Function to check if a file or directory exists in a case-sensitive manner
+check_case_sensitive_exists() {
+    local path="$1"
+    local dir
+    local base
+
+    dir=$(dirname "$path")
+    base=$(basename "$path")
+
+    if [ -e "$dir" ]; then
+        if (cd "$dir" && find . -maxdepth 1 -name "$base" | grep -q "$base"); then
```

Comment:
Could we use `find "$dir"` to save a child shell?

Reply:
Can you show me what you mean? Thanks!

Reply:
We can rewrite the cmd into `find "$dir" -maxdepth 1 -name "$base" ...` so we can avoid using a sub shell wrapped with `()`

Reply:
Is `| grep -q $base` any different from redirecting like `>/dev/null 2>&1`? I believe `find` should already filter out the path;

Reply:
> Is `| grep -q $base` any different from redirecting like `>/dev/null 2>&1`? I believe `find` should already filter out the path;

The method I used was tested in multiple different scenarios. Please ensure whatever method you suggest is still able to efficiently and reliably detect case-sensitivity regardless of the case-sensitivity of the filesystem.

Reply:
Yeah I don't know, I think I see now what `-iname` does on case insensitive file systems, which explains the `grep`.

Reply:
> Yeah I don't know, I think I see now what `-iname` does on case insensitive file systems, which explains the `grep`.

So should I leave as-is?

Reply:
Oops, I missed this notification. Yeah, leave it as it is. My recommendation came from a misunderstanding